src/framework/tooltip/Tooltip.jsx

```diff
@@ -0,0 +1,134 @@
+import React, {
+  Component,
+  PropTypes,
+} from 'react';
+
+import ReactDOM from 'react-dom';
+
+import {
+  Box,
+  Text,
+} from '../../framework/framework';
+
+export default class Tooltip extends Component {
+  constructor(props) {
+    super(props);
+
+    this.content = this.props.message;
+    this.children = this.props.children;
+    this.onMouseEnter = this.onMouseEnter.bind(this);
+    this.onMouseLeave = this.onMouseLeave.bind(this);
+  }
+
+  componentDidMount() {
+    this.tooltip = this.refs.tooltipPosition;
+  }
+
+  onMouseEnter(item) {
+    this.renderTooltip(item);
+  }
+
+  onMouseLeave() {
+    this.tooltip.style.display = 'none';
+    this.tooltip.style.opacity = 1;
+  }
+
+  getTooltipDimension() {
+    const tooltip = document.getElementById('tooltip');
+
+    let lx;
+    let ly;
+    let element = tooltip;
+    for (lx = 0, ly = 0;
+      element != null;
+      lx += element.offsetLeft,
+      ly += element.offsetTop,
+      element = element.offsetParent);
+
+    return {
+      height: tooltip.clientHeight,
+      posX: lx,
+      posY: ly,
+      width: tooltip.clientWidth,
+    };
+  }
+
+  renderTooltipLabel() {
+    return (
+      <Box roundedCorners>
+        <div id="tooltip" className="tooltip">
+          <Text>
+            {this.content}
+          </Text>
+        </div>
+      </Box>
+    );
+  }
+
+  renderTooltip(item) {
+    this.tooltip.style.display = 'inline-block';
+    this.tooltip.style.left = '0px';
+    this.tooltip.style.top = '0px';
+    this.tooltip.style.height = this.props.height;
+    this.tooltip.style.width = this.props.width;
+    let styles = 'tooltipPosition ';
+
+    const windowLimits = {
+      height: window.innerHeight,
+      width: window.innerWidth,
+    };
+
+    ReactDOM.render(
+      this.renderTooltipLabel(item),
+      this.tooltip
+    );
+
+    const tooltipDimension = this.getTooltipDimension();
+    let tooltipX = (document.getElementById('tooltipParent').clientWidth / 2) -
+      (tooltipDimension.width - 15);
+    if (tooltipDimension.posX - tooltipDimension.width < 0) {
+      tooltipX = 0;
+      styles += ' tooltipLeft';
+    }
+
+    let tooltipY = document.getElementById('tooltipParent').clientHeight;
```

Comment:
Everytime that you use $(element) it doesn't only get the element, it also converts it to a jquery object. It doesn't make sense to get the element, convert it to a jquery object and get the height with a function from that jquery object when you can just have the height from the element.